golang/cosmos/x/swingset/swingset.go

```diff
@@ -0,0 +1,75 @@
+package swingset
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Agoric/agoric-sdk/golang/cosmos/vm"
+	vstoragetypes "github.com/Agoric/agoric-sdk/golang/cosmos/x/vstorage/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// portHandler implements a vm.PortHandler.
+type portHandler struct {
+	keeper Keeper
+}
+
+type swingsetMessage struct {
+	Type    string          `json:"type"`
+	Request json.RawMessage `json:"request"`
+}
+
+type swingsetStoreSetRequest struct {
+	Entries []json.RawMessage `json:"entries"`
+}
```

Comment:
Looking at all the modules, vstorage structures inbound messages like `{ Method: string, Args: []json.RawMessage }` while the others structure them like `{ Type: string, …specific fields… }` (and vibc calls out that type corresponds with IBC_METHOD). I think the vstorage approach best corresponds with what we do elsewhere (such as "methargs" in vat deliveries) and with what's actually going on here.

```suggestion
type swingsetMessage struct {
	Method string            `json:"method"`
	Args   []json.RawMessage `json:"args"`
}
```
```patch
 	case SWINGSET_STORE_SET:
-		var request swingsetStoreSetRequest
-
-		err := json.Unmarshal(msg.Request, &request)
-		if err != nil {
-			return "", err
-		}
-		return ph.handleStoreSet(ctx.Context, request.Entries)
+		return ph.handleStoreSet(ctx.Context, msg.Args)
```
```patch
       chainSend(
         portNums.swingset,
         stringify({
-          type: 'SWINGSET_STORE_SET',
-          request: { entries },
+          method: 'SWINGSET_STORE_SET',
+          args: entries,
         }),
       );
```